### PR TITLE
remove opencensus from carbonreceiver and wavefrontreceiver

### LINKED ISSUE #20759: [receiver/carbon] Migrate from OpenCensus data model to pdata
The receiver still uses OpenCensus data model to create the metrics and converts them into pdata before sending them to the next consumer. This introduces a performance impact. The receiver should use pdata model to create the metrics instead.

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Remove the opencensus data model from carbonreceiver and wavefrontreceiver, translate directly to pmetric data.

I meant to only fix carbonreceiver, but turns out wavefrontreceiver depends on carbonreceiver for this conversion, so it was dragged in.

**Link to tracking Issue:**
Closes #20759 and #20761

**Testing:**
All tests pass.

**Documentation:**
N/A

### CODE REVIEW

File: .chloggen/use-pdata-with-carbonreceiver.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```
Comment (codeboten) [inline .chloggen/use-pdata-with-carbonreceiver.yaml:6]: Isn't this a breaking change for anyone who was using the changed interfaces?

**Review by codeboten — COMMENTED** [review]
Thanks for taking this on @atoulme! Do you think it would be much work to change the format of the tests to support metrics_golden style tests?

File: .chloggen/use-pdata-with-carbonreceiver.yaml (hunk 1 above)
Comment (atoulme) [reply .chloggen/use-pdata-with-carbonreceiver.yaml:6]: It’s not since the inputs, outputs and configs are unchanged. We keep all of it the same.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**